ngx_http_sxg_filter_module.c

```diff
@@ -45,6 +46,9 @@ typedef struct {
   ngx_str_t as;
 } ngx_subresource_t;
 
+// A context structure which is allcated for every single client request.
+// This structure will be concurrently accessed to handle multiple subrequest
+// handling.
```

Comment:
Sorry the lifetime of this struct (and lock) still confuses me.  Is it allocated once per request then reused for each subresource?